src/deviceAPI/DeviceAPI.js

```diff
@@ -0,0 +1,56 @@
+const { API } = require('aws-amplify')
+const { DEFAULT_REQ_OPTS } = require('../utils/defaultReqOpts')
+const RegexUtils = require('../utils/RegexUtils')
+
+/**
+ * Stores the identity of the device invoking functions througout the device-api service
+ *
+ * @class DeviceAPI
+ * @abstract
+ */
+class DeviceAPI {
+
+  static setIdentity(deviceId) {
+    if (!deviceId) {
+      throw Error('No Device ID specified')
+    }
+    if (!deviceId.match(RegexUtils.UUIDV4_REGEX_STRING)) {
+      throw Error(`${deviceId} is not a valid UUIDv4`)
```

Comment:
`not a valid UUID` as the error message will be more precise because we do not check if the `deviceId` is a valid uuidv4. There are more rules on what a uuidv4 (compared to a uuid of a previous version). Because we check only against a regex, we could confuse a developer here who works with the API